### PR TITLE
Change lexical scope Features

### PR DESCRIPTION
**Pull request by haslinghuis**

### CODE REVIEW

File: src/js/Features.js [hunk 1]
```diff
@@ -165,23 +165,17 @@ Features.prototype.generateElements = function (featuresElements) {
             if (!self._features[i].hideName) {
                 featureName = `<td><div>${self._features[i].name}</div></td>`;
             }
+            const newElement = $(`<tr><td><input class="feature toggle" id="feature-${i}" \
+                " name="${self._features[i].name}" title="${self._features[i].name}" type= \
+                "checkbox"/></td><td><div>${featureName} \
+                </div><span class="xs" i18n="feature${self._features[i].name}"></span></td> \
+                <td><span class="sm-min" i18n="feature${self._features[i].name}"></span>' \
+                ${feature_tip_html}</td></tr>`);
```
Comment (McGiverGim) [inline src/js/Features.js:173]: I think that the back slash `\` is not needed. The backtick ` marks the use of "template" literals and they let the use of multiline directly: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals#Multi-line_strings

File: src/js/Features.js [hunk 2]
```diff
@@ -165,23 +165,17 @@ Features.prototype.generateElements = function (featuresElements) {
             if (!self._features[i].hideName) {
                 featureName = `<td><div>${self._features[i].name}</div></td>`;
             }
+            const newElement = $(`<tr><td><input class="feature toggle" id="feature-${i}" \
+                " name="${self._features[i].name}" title="${self._features[i].name}" type= \
```
Comment (McGiverGim) [inline src/js/Features.js:169]: I think this `"` starting is not needed, the id is closed in the earlier line.

**Review by McGiverGim — CHANGES_REQUESTED** [review]

File: src/js/Features.js (hunk 2 above)
Comment (haslinghuis) [reply src/js/Features.js]: Thanks. Good spot! Refactored and tested.

File: src/js/Features.js [hunk 3]
```diff
@@ -130,58 +130,57 @@ Features.prototype.isEnabled = function (featureName) {
...
+  <td>
+    <div>${featureName}</div>
+    <span class="xs" i18n="feature${rawFeatureName}"></span>
+  </td>
+  <td>
+    <span class="sm-min" i18n="feature${rawFeatureName}"></span>
+    ${feature_tip_html}
+  </td>
+</tr>`
+            );
```
Comment (McGiverGim) [inline src/js/Features.js:178]: This indentation is ugly, but I understand the reason to do this. I suppose we can try to include in a future pr the https://www.npmjs.com/package/common-tags package to work with html code easily.

**Review by McGiverGim — APPROVED** [review]
Approved, only one comment to remember us for a future...

File: src/js/Features.js (hunk 3 above)
Comment (haslinghuis) [reply src/js/Features.js:178]: @McGiverGim In chrome devtools I see the ↵ character in innerHtml as in innerText. With template literals you have to start from column 0. Ultimately we should convert this is into Vue templates?

Comment (mikeller) [reply src/js/Features.js:178]: Yeah, this is truly ugly.
An alternative to converting this to vue that is less involved is to stick the html as a hidden element into the .html file, and then use JQuery to clone it and add the right attributes.

File: src/js/Features.js [hunk 4]
```diff
@@ -130,58 +130,50 @@ Features.prototype.isEnabled = function (featureName) {
...
+            element += `name="${self._features[i].name}" title="${self._features[i].name}"`;
+            element += `type="checkbox"/></td><td><div>${featureName}</div>`;
+            element += `<span class="xs" i18n="feature${self._features[i].name}"></span></td>`;
+            element += `<td><span class="sm-min" i18n="feature${self._features[i].name}"></span>`;
+            element += `${feature_tip_html}</td></tr>`;
+
+            const newElement = $(element);
 
-            var feature_e = newElement.find('input.feature');
+            const featureElem = newElement.find('input.feature');
```
Comment (mikeller) [inline src/js/Features.js:173]: Should this be `featureElement` as well?

File: src/js/Features.js [hunk 5]
```diff
@@ -130,58 +130,50 @@ Features.prototype.isEnabled = function (featureName) {
...
             }
-
-            var newElement = $('<option class="feature" id="feature-'
-                + i
-                + '" name="'
-                + self._features[i].name
-                + '" value="'
-                + self._features[i].bit
-                + '" i18n="feature' + self._features[i].name + '" />');
+            const newElement = $(` <option class="feature" id="feature-${i}" name="${rawFeatureName}" value="${featureBit}" i18n="feature${rawFeatureName}" />`);
```
Comment (mikeller) [inline src/js/Features.js:154]: Stray space at the start of the string. :space_invader:

Comment (haslinghuis) [reply src/js/Features.js]: 👽

**Comment by haslinghuis** [comment]
Fixed!

File: src/js/Features.js (hunk 4 above)
Comment (haslinghuis) [reply src/js/Features.js]: Yeah to be consistent in naming. And it doesn't bite the other binding (used as function argument on line 206)

**Review by mikeller — APPROVED** [review]

**Pull request merged**